src/main/kotlin/no/nav/familie/ef/iverksett/tilbakekreving/TilbakekrevingListener.kt

```diff
@@ -0,0 +1,58 @@
+package no.nav.familie.ef.iverksett.tilbakekreving
+
+import com.fasterxml.jackson.module.kotlin.readValue
+import no.nav.familie.ef.iverksett.felles.FamilieIntegrasjonerClient
+import no.nav.familie.ef.iverksett.iverksetting.IverksettingRepository
+import no.nav.familie.kontrakter.felles.arbeidsfordeling.Enhet
+import no.nav.familie.kontrakter.felles.objectMapper
+import no.nav.familie.kontrakter.felles.tilbakekreving.HentFagsystemsbehandlingRequest
+import no.nav.familie.kontrakter.felles.tilbakekreving.Ytelsestype
+import no.nav.familie.log.mdc.MDCConstants
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.slf4j.LoggerFactory
+import org.slf4j.MDC
+import org.springframework.kafka.annotation.KafkaListener
+import org.springframework.kafka.listener.ConsumerSeekAware
+import org.springframework.stereotype.Component
+import java.util.UUID
+
+@Component
+class TilbakekrevingListener(
+        private val iverksettingRepository: IverksettingRepository,
+        private val familieIntegrasjonerClient: FamilieIntegrasjonerClient,
+        private val tilbakekrevingProducer: TilbakekrevingProducer
+) : ConsumerSeekAware {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+    private val secureLogger = LoggerFactory.getLogger("secureLogger")
+
+    @KafkaListener(id = "familie-ef-iverksett",
+                   topics = ["teamfamilie.privat-tbk-hentfagsystemsbehandling-request-topic"],
+                   containerFactory = "concurrentTilbakekrevingListenerContainerFactory")
+    fun listen(consumerRecord: ConsumerRecord<String, String>) {
+        val key: String = consumerRecord.key()
+        val data: String = consumerRecord.value()
+        try {
+            MDC.put(MDCConstants.MDC_CALL_ID, UUID.randomUUID().toString())
+            transformerOgSend(data)
+        } catch (e: Exception) {
+            logger.error("Feil ved håndtering av HentFagsystemsbehandlingRequest med eksternId=$key")
+            secureLogger.error("Feil ved håndtering av HentFagsystemsbehandlingRequest med consumerRecord=$consumerRecord")
+            throw e
+        } finally {
+            MDC.remove(MDCConstants.MDC_CALL_ID)
+        }
+    }
+
+    private fun transformerOgSend(data: String) {
+        val request: HentFagsystemsbehandlingRequest =
+                objectMapper.readValue(data)
+        if (request.ytelsestype != Ytelsestype.OVERGANGSSTØNAD) {
```

Comment:
Hvorfor? familie-tilbake trenger informasjon for alle slags EF ytelser

Reply:
Vi har ikke kommet så langt enda, men kan jo legge til sjekk for alle ..